pytests/test_linearoperator.py

```diff
@@ -72,7 +71,7 @@ def test_scaled(par):
 def test_scipyop(par):
     """Verify interaction between pylops and scipy Linear operators"""
 
-    class spDiag(spLinearOperator):
+    class spDiag(LinearOperator):
```

Comment:
I also didn't want to change this, I just made it to pass the tests, I would have discussed the same after you had reviewed the PR.
Now it's fine that we use a `_LinearOperator` class as a metaclass.

Reply:
Ok. For the future, it is best to leave a test fail and explain why than do a quick fix that it may have gone unnoticed

Reply:
```
def handle_dims_dimsd(self, A: LinearOperator, B: LinearOperator):
       if self.dims != B.dims:
            B.dims = self.dims
       if self.dimsd != A.dimsd:
            A.dimsd = self.dimsd
```
A method like this could also fix our problem we can write a method that checks the dims and dimsd of `_ProductLinearOperator` and updates the dims,dimsd of A and B